### DISCUSSION TITLE
How to get window.origin in a config file nuxt3

### DISCUSSION THREAD

**Discussion by tintin10q** [nuxt/nuxt#17672] [Questions]
Hi,

I have a config directory with files like google.js which define mostly hardcoded values for google oauth2. Like this:

```
const client_id = "24...jhj0u4v.apps.googleusercontent.com";
const redirect_endpoint = "/google_callback/"; 
const redirect_uri = window.origin + redirect_endpoint;
const scope = "https://www.googleapis.com/auth/youtube.readonly https://www.googleapis.com/auth/youtube email https://www.googleapis.com/auth/youtube.readonly email";

export {
    client_id,
    redirect_endpoint,
    redirect_uri,
    scope,
};
```

In various places in my pinia store I import from this file.

I would like the window.origin to always be the origin of the application that way I don't have to hardcode it and change it everytime it changes. However I get ` window is not defined` I understant that window is not defined on the server.

How can I get the window.origin on the client and get rid of this error?  I would like to wrap this code in something like <client-only></client-only> or something. I also tried moving these files to the plugins directory and naming them ...client.js like google.client.js but it did not work.

**Comment by viii47** [comment]
If you want to set it through a component or page, then you have to use the  `onMounted(() => {})` / `mounted() {}` lifecycle hook. If it's in a plugin you have to check if the client process is active `if (process.client) { ... }` to get access to the window.

**Reply by tintin10q** [reply]
Oh so I could make the config a composable? But that that also work in a pinia store? Maybe I should just make the config a pinia store. Thank you

**Reply by viii47** [reply]
> Oh so I could make the config a composable? But that that also work in a pinia store? Maybe I should just make the config a pinia store. Thank you

Yes you could definitely make the config a composable *or* set & read the values of the config from there (given that you do need the clients *availability* in the given context). I'm not familiar with how pinia works but this should work.
